### ISSUE TITLE
Expand Azure CNI Overlay Pod CIDR after cluster creation

### ISSUE THREAD

**Issue by chasewilson** [Azure/AKS#4591]
## GA ETA*: Q1 2026
*ETAs are estimations and subject to change

Sometimes customers will create an Overlay Cluster with a Pod CIDR big enough for their current workloads and as their requirements grow, they need addition IP space. 

Currently, to address this issue customers need to recreate clusters with a bigger Pod CIDR space. 

This feature allows customers to add additional Pod CIDR subnets to an existing Azure CNI Overlay cluster without having to recreate.

**Comment by aido123** [comment]
+1

**Comment by KrzysztofSzyntar** [comment]
can you confirm the ETA - is it still Q1 2025 ?

**Comment by paulgmiller** [comment]
@robogatikov will be taking this

**Comment by hakman** [comment]
still relevant, looking forward to this feature 🙂

**Comment by asubmani** [comment]
we user CNI in our AKS clusters (from 2019).. The subnet provides I.P for both nodes and pods. Will this feature help us to switch to CNI overlay where only the nodes need i.ps from the subnet? AFAIK CNI Overlay only needs IP address space for nodes and not the pods.

**Comment by olsenme** [comment]
@chasewilson  any updates?

**Comment by paulgmiller** [comment]
@asubmani  this won't help you move from pod subnet to overlay. Still have to recreate for that. 

While alot of work was done on this we're stilled debating if we force an reimage/rollling reimage of nodes or do another substantial peice of work to make this happen without a rolling reimage.

**Comment by paulgmiller** [comment]
@asubmani  mis read your post. seems like you're using node subnet which can upgrade to overlay
https://learn.microsoft.com/en-us/azure/aks/upgrade-azure-cni#azure-cni-cluster-upgrade

At the time you upgrade to overlay you can specify whatever pod cidr you want.

Since podcidr expansion in overlay is delayed were looking tomake it so when upgrade from kubenet you can expand your cidr at kubenet -> overlay upgrade time

**Comment by asubmani** [comment]
> [@asubmani](https://github.com/asubmani) mis read your post. seems like you're using node subnet which can upgrade to overlay https://learn.microsoft.com/en-us/azure/aks/upgrade-azure-cni#azure-cni-cluster-upgrade
> We are using the old CNI network type from when AKS used to support only `kubelet` and `CNI`. Both nodes and pods get i.ps from the same subnet.

**Comment by palma21** [comment]
@asubmani yes, as Paul mentioned, the way forward for you is to upgrade from any legacy combination to overlay (legacy will eventually be retired) and in overlay this feature will allow you to adapt, but maybe you won't need it because at migration you can already adapt as well

**Comment by asubmani** [comment]
> ....maybe you won't need it because at migration you can already adapt as well
Could you elaborate on this? Are you talking about "azure level backend migration" or user initiated one? 
Given the current status of my cluster... will `az aks update ...--network-plugin-mode overlay` work? I am asking as migration to  me means spinning up a new cluster and moving my apps to the new one from the old up. Hopefully I can do an inplace migration

Sharing output of : ` az aks show -g $RG -n $AKS --query networkProfile`

![Image](https://github.com/user-attachments/assets/22453759-f409-456c-b08c-87964c99b53c)

**Comment by paulgmiller** [comment]
since network plugin mode is null you are on node subnet and can set your pod cidr when you upgrade today to be whatever you want.

**Comment by danbosscher** [comment]
Hi folks - we've unfortunately run into an unexpected blocker with this feature. We're now aiming to have this ready in public preview in Q3 and will update this post once we have a firm date lined up.

**Comment by hakman** [comment]
@danbosscher I know it's not the topic here, but could at least something like suggested in https://github.com/Azure/AKS/issues/4367 be done, or allow to disable pre-allocation?

**Comment by danbosscher** [comment]
Hi folks - this feature is in active development and we estimate to have this in Public Preview for you in October 2025.

**Comment by sf-msft** [comment]
This is now available as public preview for k8s 1.33 and above. The rest will resume with backports in January.

**Comment by mblixter** [comment]
@sf-msft Thanks for the update. 
For anyone looking for how to do this, the guide can be found here:
https://learn.microsoft.com/en-us/azure/aks/azure-cni-overlay-pod-expand

**Comment by msftjonw** [comment]
GA ETA: April '26.

**Comment by Jeff-Snowden** [comment]
@chasewilson  we're beginning to get support cases where customers are attempting this and while the --pod CIDR update command seems to work, the updated network configuration does not get passed down to the worker nodes.  

Performing a node reboot seems to correct the issue and pulls the updated CIDR.  

If you would like more context around the support cases, I can reach out to you on teams. 

@amargherio    for awareness

**Comment by sf-msft** [comment]
There was an issue with a race condition that was rolled out already - eng is investigating this now

**Comment by sf-msft** [comment]
Closing as completed and GAed
